### DISCUSSION TITLE
Inputs - by Bluetooth?

### DISCUSSION THREAD

**Discussion by Special-Niewbie** [thefloppydriver/pc2drc#5] [Q&A] [answered]
@thefloppydriver , as I know and from my knowledge with my past development mod in Wii U , the input of the Pad it's management by the BCM20702 Bluetooth, and all the related stream screen by the 5Ghz WiFi.

When I watch your video, I cannot see that you are using the rest of the inputs controllers. 

So, now I have one doubt that I would like to share with you, but pc2drc it's fully functional all the input as well (like the A,B,Y,X and so on) or just the screen as it's handle by the 5Ghz WiFi?

**Comment by thefloppydriver** [comment, marked as answer]
@Special-Niewbie I can assure you that everything on the Gamepad is handled over the WiFi connection and that the gamepad has no Bluetooth module. It [input] even works without the TSF patch. I have buttons bound to keyboard and mouse keys and I'm currently working on a bug with the gamepad input -> /dev/uinput feeder. It worked on Ubuntu 18.04 back on the old project (I played super Mario sunshine using dolphin and the Gamepad controls) If you look on the YouTube video that I have up you'll see that around 0:12 I put my hand over to the left ZL button on the Gamepad and press it, which I have bound to left click in drcvncclient (when set to mouse mode)

**Comment by thefloppydriver** [comment]
If you have any other questions or need confirmation on this, feel free to comment on the issue.

**Comment by Special-Niewbie** [comment]
@thefloppydriver if it's this the case I need to say sorry about that, because in between many devs for the Wii U scenario it's knows that the input of the Wii U it's handled by Bluetooth.... Now you make me discover something that I was sure till yesterday that was a strong knowledge base about it. In this case the Bluetooth specs that included for Wii U PAD, for what?

**Comment by thefloppydriver** [comment]
@Special-Niewbie Bluetooth is used for the WiiU Pro Controller and the Wiimotes, that's why you can pair them to your computer over bluetooth :) I hope that clears everything up for you.

**Comment by Special-Niewbie** [comment]
> @Special-Niewbie Bluetooth is used for the WiiU Pro Controller and the Wiimotes, that's why you can pair them to your computer over bluetooth :) I hope that clears everything up for you.

Yes, sorry I explained bad my self the Wii U BASE yes it need to connect other controllers, but the Wii U PAD (as you can check around) always it's written or if you need to replace it,  the specs of the Wireless antenna it's WiFi and Bluetooth and into the PAD there are 3 antennas WiFi, Bluetooth and NFC, right?

**Comment by thefloppydriver** [comment]
> > @Special-Niewbie Bluetooth is used for the WiiU Pro Controller and the Wiimotes, that's why you can pair them to your computer over bluetooth :) I hope that clears everything up for you.
> 
> Yes, sorry I explained bad my self the Wii U BASE yes it need to connect other controllers, but the Wii U PAD (as you can check around) always it's written or if you need to replace it, the specs of the Wireless antenna it's WiFi and Bluetooth and into the PAD there are 3 antennas WiFi, Bluetooth and NFC, right?

Nope, The PAD only has 2 antennas; WiFi and NFC. Bluetooth wasn't needed on the PAD so Nintendo didn't include it.

EDIT: Here is an overview of the internal structure of the pad from the now-defunct libdrc.org.
![image](https://user-images.githubusercontent.com/42722940/157511930-31538615-0b05-45e2-aa7d-43eb94d1101b.png)

**Comment by thefloppydriver** [comment]
@Special-Niewbie
![Screenshot from 2022-03-09 13-46-09](https://user-images.githubusercontent.com/42722940/157510151-04746960-f680-422d-aea2-f7eb964c75fb.png)
This is an old screenshot from libdrc.org, which isn't up anymore :( 
As you can see, the input is streamed over WiFi on port 50122. Which I can confirm as I have seen and decoded the inputs sent over the WiFi interface (without bluetooth)

Also I would really recommend buying the same WiFi adapter as me, because it's the only one that I can guarantee works perfectly with the gamepad.
